source/Octopus.Tentacle/Kubernetes/KubernetesRawScriptPodCreator.cs

```diff
@@ -80,7 +80,24 @@ protected override IList<V1Volume> CreateVolumes(StartKubernetesScriptCommandV1
                     {
                         ClaimName = KubernetesConfig.PodVolumeClaimName
                     }
-                }
+                },
+                new()
+                {
+                    Name = "agent-upgrade",
+                    Secret = new V1SecretVolumeSource
+                    {
+                        SecretName = "agent-upgrade-secret",
+                        Items = new List<V1KeyToPath>()
+                        {
+                            new()
+                            {
+                                Key = ".dockerconfigjson",
+                                Path = "config.json"
+                            }
+                        },
+                        Optional = true,
+                    },
+                },
```

Comment:
Can we define this as a method in the base `KubernetesScriptPodCreator`, so we don't have it defined twice?